cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c

```diff
@@ -0,0 +1,649 @@
+        }
+    }
+}
+
+static int _write(ieee802154_dev_t *dev, const iolist_t *iolist)
+{
+    (void)dev;
+
+    return cc26x2_cc13x2_rf_write(iolist);
+}
+
+static int _read(ieee802154_dev_t *dev, void *buf, size_t max_size,
+                 ieee802154_rx_info_t *info)
+{
+    (void)dev;
+    rfc_data_entry_general_t *entry;
+    uint8_t *psdu;
+    uint16_t pktlen;
+
+    /* shouldn't be null as we informed about a packet */
+    entry = cc26x2_cc13x2_rf_recv();
+    if (entry == NULL) {
+        return -ENOENT;
+    }
+
+    /* first two bytes are PSDU length */
+    pktlen = *(uint16_t *)&entry->data;
+    if (pktlen < CC26X2_CC13X2_RF_RX_STATUS_LEN ||
+        pktlen > IEEE802154G_FRAME_LEN_MAX + CC26X2_CC13X2_RF_RX_STATUS_LEN) {
+        entry->status = RFC_DATA_ENTRY_PENDING;
+        return -ENOENT;
+    }
+    pktlen -= CC26X2_CC13X2_RF_RX_STATUS_LEN;
+    DEBUG("[cc26x2_cc13x2_rf]: reading packet of length %i\n", pktlen);
+
+    psdu = (&entry->data) + sizeof(uint16_t);
+
+    if (max_size < pktlen) {
+        DEBUG("[cc26x2_cc13x2_rf]: buffer is too small\n");
+        entry->status = RFC_DATA_ENTRY_PENDING;
+        return -ENOBUFS;
+    }
+    else {
+        if (info != NULL) {
+            /* RSSI resides at the end of the PSDU */
+            DEBUG("RSSI: %i\n", (int)(int8_t)psdu[pktlen]);
+            info->rssi = ieee802154_dbm_to_rssi((int8_t)psdu[pktlen]);
+            info->lqi = 0;
+        }
+        memcpy(buf, psdu, pktlen);
+        entry->status = RFC_DATA_ENTRY_PENDING;
+    }
+
+    return pktlen;
+}
+
+static int _len(ieee802154_dev_t *dev)
+{
+    (void)dev;
+
+    /* shouldn't be null as we informed about a packet */
+    rfc_data_entry_general_t *entry = cc26x2_cc13x2_rf_recv();
+    assert(entry != NULL);
+
+    uint16_t pktlen = *(uint16_t *)&entry->data;
+    if (pktlen < CC26X2_CC13X2_RF_RX_STATUS_LEN) {
+        entry->status = RFC_DATA_ENTRY_PENDING;
+        return -ENOENT;
+    }
+
+    return pktlen - CC26X2_CC13X2_RF_RX_STATUS_LEN;
+}
+
+static int _off(ieee802154_dev_t *dev)
+{
+    (void)dev;
+    return -ENOTSUP;
+}
+
+static int _request_op(ieee802154_dev_t *dev, ieee802154_hal_op_t op, void *ctx)
+{
+    (void)dev;
+
+    int res = -EBUSY;
+    int state = STATE_IDLE;
+
+    switch (op) {
+    case IEEE802154_HAL_OP_TRANSMIT:
+        if (_cfg.ifs) {
+            DEBUG("[cc26x2_cc13x2_rf]: waiting IFS period\n");
+            goto end;
+        }
+        state = STATE_TX;
+        _tx_state = TX_STATE_TX;
+
+        if (cc26x2_cc13x2_rf_rx_is_on()) {
+            cc26x2_cc13x2_rf_rx_stop();
+        }
+
+        RFC_DBELL_NONBUF->RFCPEIFG = ~CPE_IRQ_LAST_COMMAND_DONE;
+        RFC_DBELL_NONBUF->RFCPEIEN |= CPE_IRQ_LAST_COMMAND_DONE;
+
+        if ((res = cc26x2_cc13x2_rf_request_transmit()) != 0) {
+            DEBUG("[cc26x2_cc13x2_rf]: failed to transmit\n");
+            RFC_DBELL_NONBUF->RFCPEIFG = ~CPE_IRQ_LAST_COMMAND_DONE;
+        }
```

Comment:
I was trying keep in a loop the transmit process until the Radio cpu returns a  `CPE_IRQ_LAST_COMMAND_DONE` interruption. This would keep alive the network interface during TX/RX operations. 
```suggestion
        while ((res = cc26x2_cc13x2_rf_request_transmit()) != 0) {
            DEBUG("[cc26x2_cc13x2_rf]: failed to transmit\n");
            RFC_DBELL_NONBUF->RFCPEIFG != ~CPE_IRQ_LAST_COMMAND_DONE;
            RFC_DBELL_NONBUF->RFCPEIEN |= CPE_IRQ_LAST_COMMAND_DONE;
        }
```

Reply:
This is unrelated to the buffer to small issue though, would be great to check on `cc26x2_cc13x2_rf_request_transmit` what is failing.

Reply:
oh, you are right, this refers to the "failed to transmit" issue, here it's repeating the process to trasmit until obtain a response from the radio CPU. Respecting to the upper layer i saw an issue generated after this error is  printed.
```
2022-10-26 15:02:03,412 # gnrc_netif: can't queue packet for sending
2022-10-26 15:02:03,415 # gnrc_netif: can't queue packet for sending
2022-10-26 15:02:04,412 # gnrc_netif: can't queue packet for sending
```
So, maybe could be presented in `sys/net/gnrc/netif/gnrc_netif.c` between the lines 1844 and 1862